tensorly/tenalg/core_tenalg/_batched_tensor_dot.py

```diff
@@ -0,0 +1,64 @@
+from ..tenalg_utils import _validate_contraction_modes
+import tensorly as tl
+from math import prod
+
+
+def batched_tensor_dot(tensor1, tensor2, modes, batched_modes=()):
```

Comment:
I think we should change its name to `batched_tensordot` to both be consistent with `tenordot` and to differentiate it with the removed`tl.tenalg.batched_tensor_dot` which was an outer product.

Reply:
I'm conflicted - I have implemented other version for factorized tensors, e.g. tensor_dot_tucker or tucker_dot_tucker. It would be more consistent to use `x_dot_y` everywhere. I don't think differentiating with the previous one is necessary - we can document the change and it wasn't used anywhere in the library anyway.

However, being consistent with Numpy might be reason enough. If we go for batched_tensordot we could actually consider just calling it `tensordot`, with an additional batched_modes parameter.